.github/workflows/tpv2.yml

```diff
@@ -16,53 +16,206 @@
 # under the License.
 name: Lint and Test Experimental Traffic Portal
 
+env:
+  # alpine:3.13
+  ALPINE_VERSION: sha256:08d6ca16c60fe7490c03d10dc339d9fd8ea67c6466dea8d558526b1330a85930
+
 on:
   pull_request:
     paths:
       - experimental/traffic-portal/**
       - .github/workflows/tpv2.yml
-    types: [opened, reopened, edited, synchronize]
+      - .github/actions/tpv2-integration-tests
+    types: [opened, reopened, ready_for_review, synchronize]
 
 jobs:
   build:
+    if: github.event.pull_request.draft == false
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v1
+        uses: actions/checkout@v3
 
       - name: Cache node modules
-        uses: actions/cache@v1
+        uses: actions/cache@v3
         with:
           path: ./experimental/traffic-portal/node_modules
           key: ${{ runner.os }}-node-${{ hashFiles('./experimental/traffic-portal/package-lock.json') }}
           restore-keys: |
             ${{ runner.os }}-node-
 
       - name: Node 16
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: 16.x
 
-      - name: Install latest Chrome
-        run: sudo apt-get update && sudo apt-get install google-chrome-stable
-
       - name: NPM install
         run: |
           cd experimental/traffic-portal/
-          npm ci
+          npm install
```

Comment:
`npm ci` was correct

Reply:
No, it's not. `npm ci` removes all of the `node_modules` and re-installs them fresh, effectively rendering our earlier caching pointless. Not loading the cached modules, then deleting them, then re-downloading them cuts the time it takes to lint and run unit tests down from like 5 minutes to about 1.5 minutes.

Reply:
If thats the case then `npm install` shouldn't be necessary then? Assuming it isn't then you can just add `if: steps.cache-node-modules.outputs.cache-hit != 'true'` so that it doesn't overwrite in that case.

Reply:
I did exactly that like three minutes before you made this comment.